scripts/legacy_expression_tool.md

```diff
@@ -0,0 +1,171 @@
+# Legacy Expression tool
+
+With release 3.2.0 of IoT Agent Node lib, the legacy expressions language support 
+has been removed. This folder contains the scripts to ease to migrate the legacy expressions. 
+
+A manual migration is required, because the expressions need to be converted to JEXL expressions manually.
+
+This python script will help you to migrate the legacy expressions to JEXL expressions. It will search for the legacy 
+expressions in the database and
+replace them with the new expressions.
+
+This script will search for the legacy expressions in the database and replace them with the new expressions. It can be also 
+be executed in dry-run mode, so it will not modify the database, but it will generate a series os files that can be used to 
+ease and verify data migration.
+
+## Output files
+
+The script generates 4 different files each time it is executed: 
+
+### `legacy_expression_ocurrences.json`
+
+This file contains information about the objects where the legacy expressions are found. It contains the following information 
+for each legacy expression found:
+
+- _id: The id of the object where the legacy expression is found
+- expression: The legacy expression found
+- type: The type of the object where the legacy expression is found (I.E: active.expression)
+- fiware_service: The fiware service of the object where the legacy expression is found
+- fiware_servicepath: The fiware service path of the object where the legacy expression is found
+
+### `legacy_expression_list.json`
+
+This file contains just a list of all the legacy expressions found in the database. It can be used to build the translation 
+file. The file looks like this:
+
+```json
+[
+    "${legacy-expression-1}",
+    "${legacy-expression-2}",
+    ...
+    "${legacy-expression-n}"
+]
+```
+
+### `objects_replaced.json`
+
+This file contains a list of objects where the legacy expressions have been replaced with all the information associated. This 
+file is useful to preview the changes expected to be done in the database before applying them.
+
+### `objects_backup.json`
+
+This file contains a list of objects containing legacy expression that have been backed up. This file is useful to preview the 
+changes expected to be done in the database before modifying it.
```

Comment:
```suggestion
This file contains a list of documents containing legacy expression that have been backed up. This file is useful to preview the 
changes expected to be done in the database before modifying it. This file is generated only when actual modifications are done to the DB (ie. when `--commit` parameter is used).
```

Reply:
Again, I'm not sure... please check my suggestion is correct.

Reply:
It always generates backup file, no matter if committing or not the changes to the DB. Considering to store the backup only when committing can be considered

Reply:
No problem with generating always the backup, but it should be documented (eg. "Note that this file is generated no matter if actual changes are going to be done in the DB (i.e. `--commit` is used) or not").

Reply:
done in https://github.com/telefonicaid/iotagent-node-lib/pull/1381/commits/3e7c7b1e2b5300c32d6e5a3f13b9852b370de0c6